### PR TITLE
Add triggers firing on less items than all

### PR DESCRIPTION
**Pull request by iafb**
- Add #175 Allow triggers to fire on less items than all

### CODE REVIEW

File: src/GRA.Data/Repository/TriggerRepository.cs [hunk 1]
```diff
@@ -144,48 +144,46 @@ private IQueryable<Model.Trigger> ApplyFilters(BaseFilter filter)
             // create a list of triggers to remove based on badge and challenge earnings
             var itemsToRemove = new List<Model.Trigger>();
 
-            // get a list of triggers that fire based on badge earnings
-            var badgeTriggers = triggers
-                .Where(_ => _.RequiredBadges != null && _.RequiredBadges.Count > 0);
+            // get a list of triggers that fire based on badge or challenge earnings
+            var itemTriggers = triggers.Where(_ => _.ItemsRequired > 0);
```
Comment (k7hpn) [inline src/GRA.Data/Repository/TriggerRepository.cs:148]: So this means that in order for badges and challenges to work from here on forward the `ItemsRequired` property will need to be set, am I reading that right?

**Review by k7hpn — COMMENTED** [review]
Just to make sure I understand how this works...

File: src/GRA.Data/Repository/TriggerRepository.cs (hunk 1 above)
Comment (iafb) [reply src/GRA.Data/Repository/TriggerRepository.cs:148]: The `ItemsRequired` field is a non-nullable int so it gets set to default if the user doesn't enter anything for it.

**Review by k7hpn — CHANGES_REQUESTED** [review]
I opened a PR where I made the required challenges/triggers into a panel, I think we should use the disclosure-triangle approach we use at the bottom for awards on the require challenges/triggers. What do you think?

I also added a "No challenges/triggers required" row when there are none but your magic adding stuff doesn't remove that row, can you modify your logic to handle that properly?

**Pull request merged**